loky/process_executor.py

```diff
@@ -411,9 +411,12 @@ def _process_worker(call_queue, result_queue, initializer, initargs,
         if call_item is None:
             # Notify queue management thread about clean worker shutdown
             result_queue.put(pid)
-            with worker_exit_lock:
+            is_clean = worker_exit_lock.acquire(True, timeout=timeout)
```

Comment:
Thinking back about this one, I am not sure that using `timeout` here is the right think to do.
Indeed, if timeout is small (for instance `0.1`), then this case could be hit on a busy system while the executor is still alive, and this might cause an error if the process end is detected before its `pid` is received by the `executor_manager_thread`.

So I would rather use a large value (let say 10s seems enough) as this is not critical for performances.
another way to fix this would be to use the sentinel from [`parent_process`](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.parent_process) to detect that the main thread is dead.

WDYT @haim0n ?

Reply:
The large value does make sense. How about using here the `max(timeout, 10)` ?